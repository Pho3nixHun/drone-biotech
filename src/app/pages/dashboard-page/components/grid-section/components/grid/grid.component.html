<div class="grid gap-4 grid-cols-subgrid col-span-full">
    <ng-content />
</div>

<!-- It projects <div>-s and <app-grid-item>-s. -->
