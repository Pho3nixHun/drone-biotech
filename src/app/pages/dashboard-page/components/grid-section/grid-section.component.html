<div class="bg-slate-100 p-5 rounded-lg flex flex-col space-y-4">
    <ng-content select="app-grid-section-header" />
    <ng-content />
</div>
