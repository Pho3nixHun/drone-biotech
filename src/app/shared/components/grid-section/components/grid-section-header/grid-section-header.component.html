<div class="flex flex-row justify-between">
    <ng-content select="[h3, a]" />
</div>
