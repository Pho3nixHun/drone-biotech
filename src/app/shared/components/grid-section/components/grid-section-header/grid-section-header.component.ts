import { Component } from '@angular/core';

@Component({
    selector: 'app-grid-section-header',
    imports: [],
    templateUrl: './grid-section-header.component.html',
})
export class GridSectionHeaderComponent {}
