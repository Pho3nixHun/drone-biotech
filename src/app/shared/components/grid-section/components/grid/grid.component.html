<div class="grid gap-4 col-span-full">
    <ng-content select="[app-grid-header, app-grid-item-list]" />
</div>
