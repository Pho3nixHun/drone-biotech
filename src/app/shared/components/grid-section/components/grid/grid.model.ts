export interface GridVM {
    gridColsLength: GridColsLength;
}

export enum GridColsLength {
    SIX = 'grid-cols-6',
    SEVEN = 'grid-cols-7',
}
