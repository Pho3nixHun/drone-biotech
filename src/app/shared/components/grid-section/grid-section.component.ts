import { Component } from '@angular/core';

@Component({
    selector: 'app-grid-section',
    imports: [],
    templateUrl: './grid-section.component.html',
})
export class GridSectionComponent {}
