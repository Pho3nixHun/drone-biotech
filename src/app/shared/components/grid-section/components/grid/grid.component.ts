import { Component, input } from '@angular/core';
import { GridVM } from './grid.model';
import { NgClass } from '@angular/common';

@Component({
    selector: 'app-grid',
    imports: [NgClass],
    templateUrl: './grid.component.html',
})
export class GridComponent {
    public vm = input.required<GridVM>();
}
