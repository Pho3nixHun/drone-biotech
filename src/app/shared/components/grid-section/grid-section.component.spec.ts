import { ComponentFixture, TestBed } from '@angular/core/testing';
import { Component } from '@angular/core';
import { GridComponent } from './components/grid/grid.component';
import { GridSectionHeaderComponent } from './components/grid-section-header/grid-section-header.component';
import { GridSectionComponent } from './grid-section.component';

@Component({
    imports: [GridComponent, GridSectionHeaderComponent, GridSectionComponent],
    template: `
        <app-grid-section>
            <app-grid-section-header></app-grid-section-header>
            <app-grid></app-grid>
            <p>Should not be projected</p>
        </app-grid-section>
    `,
})
class TestHostComponent {}

describe('GridSectionComponent', () => {
    let fixture: ComponentFixture<TestHostComponent>;
    let compiled: HTMLElement;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [TestHostComponent],
        }).compileComponents();

        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    // Snapshot testing
    it('should render the template correctly', () => {
        // There is no need to arrange

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});
