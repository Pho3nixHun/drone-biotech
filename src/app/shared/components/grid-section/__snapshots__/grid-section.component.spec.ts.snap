// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridSectionComponent should render the template correctly 1`] = `
<div
  id="root0"
  ng-version="19.2.9"
>
  <app-grid-section>
    <div
      class="bg-slate-100 p-5 rounded-lg flex flex-col space-y-4"
    >
      <app-grid-section-header>
        <div
          class="flex flex-row justify-between"
        />
      </app-grid-section-header>
      <app-grid
        ng-reflect-vm="[object Object]"
      >
        <div
          class="grid gap-4 col-span-full grid-cols-6"
          ng-reflect-ng-class="grid-cols-6"
        />
      </app-grid>
    </div>
  </app-grid-section>
</div>
`;
